SCClassLibrary/External/Ableton/SCClockMeterSync.sc

```diff
@@ -0,0 +1,246 @@
+// originally conceived for LinkClock (Ableton Link support)
+// but this may in theory apply to any clock that answers to
+// baseBarBeat_ (e.g. MIDISyncClocks, or BeaconClock)
+
+// TODO OSC message prefix with SC_
+
+SCClockMeterSync {
```

Comment:
naive question -- why is there is `SC` prefix on this class name?

Reply:
Link peers may include SC LinkClocks and clocks or timelines from other applications. This class doesn't (can't) try to sync any non-SC peers.

I didn't call it LinkMeterSync because I believe that it might work with the Utopia quark's BeaconClock as well. (BeaconClock promises to sync beats, but not barlines.)

Actually, I did test that just now -- BeaconClock needs to add `numPeers { ^addrBook.peers.size }` but after that, SCClockMeterSync works correctly in all respects. :open_mouth: 

So I guess I should remove references to Link from the messaging (because it *is* actually a more general solution).

Reply:
btw. why didn't you call it `ClockMeterSync`?

Reply:
> The "SC" prefix is meant to clarify that it's only for us -- SC clocks, not "clocks" in general.

that makes sense to me. thank you for taking the time to make this suitably generic! that's not fun work

> btw. why didn't you call it ClockMeterSync?

that's.. the exact question i asked.

Reply:
Btw I'd welcome suggestions for other names for this class. For the 17 years I've been using SC, I've been consistently terrible at naming classes and methods, with no sign of any improvement. If you can think of something that reads better, I'd be happy to hear it. (Sincere request... `SCClockMeterSync` is rather clunky and I wouldn't mind changing the name.)

Reply:
I think that the difficulty in naming is an objective difficulty because programming exceeds the capacity of language. The choice is either to be as explicit as possible (and look horrible) or delegate to the background knowledge of the reader (and in lucky rare cases, find an elegant expression). I would go for the former in rarely/internally used names, the latter for  everyday code.

We don't usually make it explicit that only SC is concerned, only if something is specifically for something else, so I find the SC a bit redundant. Apart from this, the name seems fine. Alternatives might be `SharedMeter` / `SharedMetre` / `SharedMetrum`, `NetMetre` if you look for something shorter.

Reply:
On further consideration, MeterSync is probably fine and leave it to documentation for the specifics. The one thing I might add to the documentation is to be clear about the fact that MeterSync assumes it's the clock's responsibility to sync beats and tempo (no TempoClock support, because a/ different machines, if beats aren't synced, you can't sync barlines at all and b/ same sclang instance, there's no reason to have two TempoClocks running identically -- just use the one).